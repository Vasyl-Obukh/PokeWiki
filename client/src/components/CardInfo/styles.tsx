import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const Info = styled.div`
  padding: 10px 10px 0;
  color: #333;
`;

export const Name = styled(Link)`
  display: block;
  margin: 0;
  font-size: 18px;
  color: #333;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  transition: all .2s;
  
  &:hover {
    color: #777;
  }
`;

export const Entry = styled.div`
  margin: 5px 0;
  text-align: center;
  color: #444;
`;

export const Key = styled.div`
  font-weight: bold;
  &::after {
    content: ':';
  }
`;

export const Values = styled.ul`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0;
  margin: 0;
`;

export const Value = styled.li`
  list-style: none;
  padding: 5px 10px;
  font-size: 14px;
  
  &:not(:last-child) {
    border-right: 2px solid lightgray;
  }
`;