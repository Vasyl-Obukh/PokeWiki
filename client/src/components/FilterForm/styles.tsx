import styled from 'styled-components';
import {FormSection} from 'redux-form';

export const Form = styled.form`
  padding: 0 50px;
`;

export const StyledFormSection = styled(FormSection)`
  display: grid;
  grid-template-columns: 1fr 1fr;
`;

export const Label = styled.label`
  margin-left: 5px;
`;

export const FormButton = styled.button`
  padding: 5px 10px;
  margin-top: 25px;
  width: 100%;
  border-radius: 20px;
  background-color: ${props => props.theme.backgroundPrimary};
  text-transform: capitalize;
  color: white;
  border: none;
  cursor: pointer;
`;