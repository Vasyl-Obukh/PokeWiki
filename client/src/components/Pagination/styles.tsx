import styled from 'styled-components';

export const List = styled.ul`
  display: flex;
  padding: 0;
  justify-content: center;
`;

export const ListItem = styled.li`
  list-style: none;
  color: white;
`;

export const Item = styled.span<{current?: boolean, to?: string, as?: object}>`
  display: flex;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  margin: 0 5px;
  align-items: center;
  justify-content: center;
  background-color: ${props => props.current ? 'white' : props.theme.backgroundPrimary};
  border: ${props => props.current ? `1px solid ${props.theme.backgroundPrimary}` : 'none' };
  color: ${props => props.current ? props.theme.backgroundPrimary : 'inherit'};
  text-decoration: none;
`;