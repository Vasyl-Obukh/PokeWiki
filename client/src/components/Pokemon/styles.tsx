import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  padding-bottom: 50px;
  flex-direction: column;
  align-items: center;
`;

export const Image = styled.img`
  width: 275px;
  height: 275px;
  border-radius: 50%;
  background-color: #f0f0f0;
  box-shadow: 0 0 5px darkslategray;
`;

export const Info = styled.div`
  display: grid;
  margin-top: 25px;
  grid-template-columns: auto auto;
  grid-column-gap: 5px;
  grid-row-gap: 10px;
  justify-content: center;
`;

export const InfoKey = styled.div`
  font-weight: bold;
  text-transform: capitalize;
  text-align: right;
  
  &::after {
    content: ':';
  }
`;

export const InfoValue = styled.div`
  font-style: italic;
  font-weight: ${props => props.primary ? 'bold' : 'normal'};
  text-transform: ${props => props.primary ? 'uppercase' : 'none'};
`;

export const ValueItem = styled.span`
  padding: 5px;
  
  &:not(:first-child) {
    border-left: 1px solid gray;
  }
`;