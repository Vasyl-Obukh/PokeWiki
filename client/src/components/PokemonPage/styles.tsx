import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 50px 15px;
  height: 100%;
`;

export const Main = styled.main`
  padding: 25px 10px;
  border-radius: 25px;
  box-shadow: 0 0 5px darkslategray;
`;