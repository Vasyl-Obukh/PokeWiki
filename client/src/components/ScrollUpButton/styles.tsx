import styled from 'styled-components';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

export const Button = styled(FontAwesomeIcon)`
  position: fixed;
  display: block;
  bottom: 150px;
  right: calc((100vw - ${props => props.theme.basicWidth}) / 2);
  height: 50px;
  padding: 10px;
  border-radius: 50%;
  background-color: ${props => props.theme.backgroundPrimary};
  font-size: 25px;
  color: white;
  cursor: pointer;

  && {
    width: 50px;
  }
`;