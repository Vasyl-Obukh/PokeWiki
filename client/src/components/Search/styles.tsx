import styled from 'styled-components';
import {Field} from 'redux-form';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';

export const Form = styled.form`
  position: relative;
`;

export const StyledField = styled(Field)`
  padding: 10px 15px 10px 35px;
  width: 150px;
  border-radius: 25px;
  border: none;
  outline: none;
  background-color: #fff;
  color: black;
  box-shadow: 0 2px 5px darkslategray;
  transition: all .5s;
  
  &:focus {
    width: 350px;
    box-shadow: -2px 2px 5px darkslategray;
  }
`;

export const SearchIcon = styled(FontAwesomeIcon)`
  position: absolute;
  left: 10px;
  top: calc(50% - 8px);
`;