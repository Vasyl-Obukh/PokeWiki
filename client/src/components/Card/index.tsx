import * as React from 'react';
import * as Styles from './styles';
import CardInfo from '../CardInfo';
import getPokemonPageUrl from '../../utils/pageUrl';

const Card = (props) => {
  const { name, thumb = '#', id, abilities = [], types = [] } = props.data;
  const url = getPokemonPageUrl(id);

  return (
    <Styles.Container>
      <Styles.ThumbWrapper to={url}>
        <Styles.Thumb src={thumb} alt={name} />
      </Styles.ThumbWrapper>
      <CardInfo data={{name, url, abilities, types}}>
      </CardInfo>
    </Styles.Container>
  );
};

export default Card;