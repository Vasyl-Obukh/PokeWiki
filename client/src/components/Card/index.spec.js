import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Card  from './index';
import * as Styles from './styles';
import configureStore from 'redux-mock-store';

Enzyme.configure({adapter: new Adapter()});

describe('<Card /> snapshot tests', () => {
  const mockStore = configureStore();
  const initialState = {cards: {data: {}}};
  const store = mockStore(initialState);
  let data;

  beforeEach(() => {
    data = {
      name: 'bulbasaur',
      thumb: '/some/url',
      id: 1,
      abilities: ['ability 1', 'ability 2'],
      types: ['poison', 'grass']
    };
  });

  test('Should render correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Card data={data} />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should not break if data has not some properties', () => {
    delete data.thumb;
    delete data.abilities;
    delete data.types;

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Card data={data} />
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Thumb url should be assigned to href attribute in img tag', () => {
    const component = mount(
      <Provider store={store}>
        <Router>
          <Card data={data} />
        </Router>
      </Provider>
    );
    expect(component.find('img').prop('src')).toBe(data.thumb);
  });

  test('Passed id should be contained in href', () => {
    const component = mount(
      <Provider store={store}>
        <Router>
          <Card data={data} />
        </Router>
      </Provider>
    );
    expect(component.find(Styles.ThumbWrapper).prop('to')).toContain(data.id);
  });
});