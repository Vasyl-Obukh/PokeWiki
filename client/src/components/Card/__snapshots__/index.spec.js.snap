// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> snapshot tests Should not break if data has not some properties 1`] = `
<div
  className="sc-bdVaJa dhyrWd"
>
  <a
    className="sc-bwzfXH zJzk"
    href="/pokemon/1"
    onClick={[Function]}
  >
    <img
      alt="bulbasaur"
      className="sc-htpNat kQspvn"
      src="#"
    />
  </a>
  <div
    className="sc-bxivhb erBfwI"
  >
    <a
      className="sc-ifAKCX gzbZPg"
      href="/pokemon/1"
      onClick={[Function]}
    >
      bulbasaur
    </a>
    <div
      className="sc-EHOje Sxglg"
    >
      <div
        className="sc-bZQynM jLRQxB"
      >
        Abilities
      </div>
      <ul
        className="sc-gzVnrw ddaANZ"
      />
    </div>
    <div
      className="sc-EHOje Sxglg"
    >
      <div
        className="sc-bZQynM jLRQxB"
      >
        Types
      </div>
      <ul
        className="sc-gzVnrw ddaANZ"
      />
    </div>
  </div>
</div>
`;

exports[`<Card /> snapshot tests Should render correctly 1`] = `
<div
  className="sc-bdVaJa dhyrWd"
>
  <a
    className="sc-bwzfXH zJzk"
    href="/pokemon/1"
    onClick={[Function]}
  >
    <img
      alt="bulbasaur"
      className="sc-htpNat kQspvn"
      src="/some/url"
    />
  </a>
  <div
    className="sc-bxivhb erBfwI"
  >
    <a
      className="sc-ifAKCX gzbZPg"
      href="/pokemon/1"
      onClick={[Function]}
    >
      bulbasaur
    </a>
    <div
      className="sc-EHOje Sxglg"
    >
      <div
        className="sc-bZQynM jLRQxB"
      >
        Abilities
      </div>
      <ul
        className="sc-gzVnrw ddaANZ"
      >
        <li
          className="sc-htoDjs jtBnBC"
        >
          ability 1
        </li>
        <li
          className="sc-htoDjs jtBnBC"
        >
          ability 2
        </li>
      </ul>
    </div>
    <div
      className="sc-EHOje Sxglg"
    >
      <div
        className="sc-bZQynM jLRQxB"
      >
        Types
      </div>
      <ul
        className="sc-gzVnrw ddaANZ"
      >
        <li
          className="sc-htoDjs jtBnBC"
        >
          poison
        </li>
        <li
          className="sc-htoDjs jtBnBC"
        >
          grass
        </li>
      </ul>
    </div>
  </div>
</div>
`;
